packages/features/form-builder/schema.ts

```diff
@@ -31,7 +33,7 @@ export const EditableSchema = z.enum([
 ]);
 
 const baseFieldSchema = z.object({
-  name: z.string(),
+  name: z.string().transform(getValidRhfFieldName),
```

Comment:
It would allow the transformation to happen in both API and if someone just intercepts the request at client and modifies the identifier.